### PR TITLE
Fix bug that accidentally disabled training op tests

### PR DESCRIPTION
**Pull request by baijumeswani**
A recent pull request #16326 accidentally disabled the training op tests from being run in the training pipeline.

This pull request fixes that.

Thanks @pengwa @PeixuanZuo for reporting the problem. Sorry for the inconvenience.

### CODE REVIEW

**Comment by baijumeswani** [comment]
This fix does not seem correct. Closing this for now.

File: cmake/onnxruntime_unittests.cmake [hunk 1]
```diff
@@ -377,27 +377,25 @@ endif()
 
 # Disable training ops test for minimal build as a lot of these depend on loading an onnx model.
 if (NOT onnxruntime_MINIMAL_BUILD)
-  if (onnxruntime_ENABLE_TRAINING_CORE)
-    file(GLOB_RECURSE orttraining_test_trainingops_cpu_src CONFIGURE_DEPENDS
-      "${ORTTRAINING_SOURCE_DIR}/test/training_ops/compare_provider_test_utils.cc"
-      "${ORTTRAINING_SOURCE_DIR}/test/training_ops/function_op_test_utils.cc"
-      "${ORTTRAINING_SOURCE_DIR}/test/training_ops/cpu/*"
-      )
+  file(GLOB_RECURSE orttraining_test_trainingops_cpu_src CONFIGURE_DEPENDS
```
Comment (pengwa) [inline cmake/onnxruntime_unittests.cmake:380]: is it better we use onnxruntime_ENABLE_TRAINING_OPS to control the op tests?

Comment (baijumeswani) [reply cmake/onnxruntime_unittests.cmake:380]: Will do.
But from what I understand, `onnxruntime_ENABLE_TRAINING_OPS` and `onnxruntime_ENABLE_TRAINING_CORE` should always be on at the same time. (when `--enable_training` flag is provided). Trying to figure out where the discrepancy comes from.

Comment (pengwa) [reply cmake/onnxruntime_unittests.cmake]: Yes, your intention is correct, while onnxruntime_ENABLE_TRAINING_CORE is undefined

**Review by pengwa — APPROVED** [review]
Confirmed the tests run now. 

2023-06-27T18:17:42.1001595Z 1: [----------] 4 tests from PadAndUnflattenTest
2023-06-27T18:17:42.1001922Z 1: [ RUN      ] PadAndUnflattenTest.FloatType1D
2023-06-27T18:17:42.1002561Z 1: 2023-06-27 18:17:42.099740932 [W:onnxruntime:Default, base_tester.cc:763 ExecuteModelForEps] registered execution providers CPUExecutionProvider were unable to run the model.
2023-06-27T18:17:42.1034317Z 1: [       OK ] PadAndUnflattenTest.FloatType1D (3 ms)
2023-06-27T18:17:42.1034928Z 1: [ RUN      ] PadAndUnflattenTest.FloatType2D
2023-06-27T18:17:42.1035710Z 1: 2023-06-27 18:17:42.103221641 [W:onnxruntime:Default, base_tester.cc:763 ExecuteModelForEps] registered execution providers CPUExecutionProvider were unable to run the model.
2023-06-27T18:17:42.1068306Z 1: [       OK ] PadAndUnflattenTest.FloatType2D (3 ms)
2023-06-27T18:17:42.1068745Z 1: [ RUN      ] PadAndUnflattenTest.MLFloat16Type1D
2023-06-27T18:17:42.1069956Z 1: 2023-06-27 18:17:42.106670451 [W:onnxruntime:Default, base_tester.cc:763 ExecuteModelForEps] registered execution providers CPUExecutionProvider were unable to run the model.
2023-06-27T18:17:42.1104736Z 1: [       OK ] PadAndUnflattenTest.MLFloat16Type1D (3 ms)
2023-06-27T18:17:42.1105176Z 1: [ RUN      ] PadAndUnflattenTest.MLFloat16Type2D
2023-06-27T18:17:42.1105865Z 1: 2023-06-27 18:17:42.110302056 [W:onnxruntime:Default, base_tester.cc:763 ExecuteModelForEps] registered execution providers CPUExecutionProvider were unable to run the model.
2023-06-27T18:17:42.1142108Z 1: [       OK ] PadAndUnflattenTest.MLFloat16Type2D (3 ms)
2023-06-27T18:17:42.1142901Z 1: [----------] 4 tests from PadAndUnflattenTest (14 ms total)

**Pull request merged**